### PR TITLE
[ELYEE-44] Update CI to also run with JDK 21

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/ELYEE-44
https://issues.redhat.com/browse/ELYEE-45

### CODE REVIEW

**Review by ivassile — APPROVED** [review]

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -9,16 +9,17 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}-jdk${{ matrix.java }}
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
+        java: ['11', '21']
```
Comment (darranl) [inline .github/workflows/pr-ci.yaml]: Should we also add 17?

Comment (fjuma) [reply .github/workflows/pr-ci.yaml]: Added JDK 17.

**Comment by fjuma** [comment]
Strange, we seem to have an issue with being able to run these jobs. Something similar also happened for our other projects:

e.g.,
https://github.com/wildfly-security/elytron-web/pull/250
https://github.com/wildfly-security/wildfly-openssl/pull/136

@PrarthonaPaul Would you have some time to take a closer look at the jobs to see if we can get them running successfully? Some examples of other projects that are running successfully with multiple JDK versions can be found here:

https://github.com/wildfly/galleon/blob/main/.github/workflows/ci.yml
https://github.com/wildfly/wildfly-client-config/blob/main/.github/workflows/ci.yml

Looking at those examples, they aren't including macos runners so we maybe we need to limit to just JDK 11 for macos or perhaps there's something else in our config that we're missing or need to tweak.

**Comment by PrarthonaPaul** [comment]
Its odd, the other projects' CI does not have any reasons listed for why the CI is failing. It just says "The Job Failed" when you go on details. 
It could be because of `os: [ubuntu-latest, windows-latest, macos-latest ]` where the other projects are not including MacOS. 
Also, I noticed this file has 
`runs-on: ${{ matrix.os }}-jdk${{ matrix.java }}` 
while the wildfly has 
`name: ${{ matrix.os }}-jdk${{ matrix.java }}
    runs-on: ${{ matrix.os }}`
Could that be something?

**Comment by fjuma** [comment]
The `name` determines how the the CI job will be listed in the output (e.g., see https://github.com/wildfly/wildfly-client-config/pull/46). The `runs-on` defines the machine the job should be run on (e.g., linux, etc.) so any differences there could be good things to look at.

**Comment by fjuma** [comment]
Superseded by https://github.com/wildfly-security/wildfly-elytron-ee/pull/32. Thanks very much @PrarthonaPaul!

**Pull request merged**